src/main/java/cs2103/v15_1j/jimjim/command/ChangeCommand.java

```diff
@@ -59,10 +68,61 @@ public LocalTime getNewEndTime() {
     }
     
 	@Override
-	public UIFeedback execute(DataLists searchResultsList, DataLists masterList, 
-							  Storage storage, Searcher searcher, Stack<UndoableCommand> undoCommandHistory) {
+	public UIFeedback undo(DataLists searchResultsList, DataLists masterList, Storage storage, Searcher searcher,
+			Stack<UndoableCommand> undoCommandHistory, Stack<UndoableCommand> redoCommandHistory) {
 		// TODO Auto-generated method stub
 		return null;
 	}
-
+	
+	@Override
+	public UIFeedback execute(DataLists searchResultsList, DataLists masterList, 
+							  Storage storage, Searcher searcher, Stack<UndoableCommand> undoCommandHistory,
+							  Stack<UndoableCommand> redoCommandHistory) {
+		try {
+			backup = masterList.getTaskEvent(taskNum, prefix);
+            DeadlineTask tempDeadlineTask = null;
+            Event tempEvent = null;
+            if (backup instanceof DeadlineTask) {
+            	tempDeadlineTask = (DeadlineTask) backup;
+            } else if (backup instanceof Event) {
+            	tempEvent = (Event) backup;
+            }
+            
+            if (newName != null) {
+            	backup.setName(newName);
+            }
+            if (newStartDate != null) {
+            	if (backup instanceof DeadlineTask) {
+            		tempDeadlineTask.setDate(newStartDate);
+            	} else if (backup instanceof Event) {
+            		EventTime currentEventTime = tempEvent.getDateTimes().get(0);
+            		currentEventTime.setStartDate(newStartDate);
+            	}
+            }
+            if (newStartTime != null) {
+            	if (backup instanceof DeadlineTask) {
+            		tempDeadlineTask.setTime(newStartTime);
+            	} else if (backup instanceof Event) {
+            		EventTime currentEventTime = tempEvent.getDateTimes().get(0);
+            		currentEventTime.setStartTime(newStartTime);
+            	}
+            }
+            if (newEndDate != null) {
+            	if (backup instanceof Event) {
+            		EventTime currentEventTime = tempEvent.getDateTimes().get(0);
+            		currentEventTime.setEndDate(newEndDate);
+            	}
+            }
+            if (newEndTime != null) {
+            	if (backup instanceof Event) {
+            		EventTime currentEventTime = tempEvent.getDateTimes().get(0);
+            		currentEventTime.setEndTime(newEndTime);
+            	}
+            }
+		} catch (IndexOutOfBoundsException e) {
+			return new FailureFeedback(
+                    "There is no item numbered " + this.prefix + this.taskNum);
+		}
+		return null;
```

Comment:
return statement should by inside try block. Should return ChangeFeedback or sth